### PR TITLE
MegaMenu: Add mega menu to new top nav design

### PR DESCRIPTION
**Pull request by torkelo**
Part of https://github.com/grafana/grafana/issues/50791 

* [x] Adds new MegaMenu component that wraps the existing NavBarMenu (shared with old nav)
   * [x]  Had to create a copy of NavBarMenu (not NavItem, still shared) in order to make the changes that are needed without complicating things. Differences:
      * Faster animation
      * Different position  (and position that depends on top search bar is visible)
      * Different background
      * Maybe we will remove the overlay (still there in this PR)
      * No exit animation 
      * Might change close button (still the same in this PR)
      * We probably will introduce other tweaks (possibly to NavItem as well)       * 
* [x] Show mega menu from TopNavPage
* [x] Change home icon from grafana logo to home icon in megamenu (only in topnav megamenu)
* [x] Fix position of MegaMenu so that it is below topnav 
* [x] Fix position of MegaMenu so that it correct when search bar is collapsed 
* [x] Tests

### CODE REVIEW

**Review by ashharrison90 — CHANGES_REQUESTED** [review]
couple of minor comments about not changing the existing `NavBarMenu` as some things break e.g. overlay doesn't fill the whole screen anymore, closing animation is gone, etc. i'll merge main so the betterer changes go away 😄

**Comment by torkelo** [comment]
@ashharrison90 thanks for reviewing! I have pushed an update that restores the NavBarMenu (overlay, exist animation)

File: public/app/core/components/MegaMenu/MegaMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        activeItem={activeItem}
+        navItems={[homeItem, ...coreItems, ...pluginItems, ...configItems]}
+        onClose={onClose}
+        searchBarHidden={searchBarHidden}
+      />
+    </div>
+  );
+});
+
+MegaMenu.displayName = 'NavBar';
```
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/core/components/MegaMenu/MegaMenu.tsx]: good catch! fixed

File: public/app/core/components/NavBar/NavBar.tsx [hunk 2]
```diff
@@ -160,12 +159,11 @@ export const NavBar = React.memo(() => {
         </NavBarContext.Provider>
       </nav>
       {showSwitcherModal && <OrgSwitcher onDismiss={toggleSwitcherModal} />}
-      {(menuOpen || menuAnimationInProgress) && (
+      {menuOpen && (
```
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/core/components/NavBar/NavBar.tsx]: fixed

File: public/app/core/components/NavBar/NavBarMenu.test.tsx [hunk 3]
```diff
@@ -16,17 +16,9 @@ jest.mock('@grafana/runtime', () => ({
...
   beforeEach(() => {
-    render(
-      <NavBarMenu
-        isOpen
-        onClose={mockOnClose}
-        navItems={mockNavItems}
-        setMenuAnimationInProgress={mockSetMenuAnimationInProgress}
-      />
-    );
+    render(<NavBarMenu isOpen onClose={mockOnClose} navItems={mockNavItems} />);
```
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/core/components/NavBar/NavBarMenu.test.tsx]: fixed

**Review by ashharrison90 — APPROVED** [review]
lgtm 🙌 

think the toggle logic isn't working correctly btw (clicking the hamburger doesn't close the menu once it's open), but we can fix that in a later PR 🔧

**Pull request merged**